Realm/ObjectServerTests/RLMFlexibleSyncServerTests.mm

```diff
@@ -909,4 +916,184 @@ - (void)testFlexibleSyncAddObjectOutsideQuery {
     }];
     [self waitForExpectations:@[ex] timeout:20];
 }
+
+- (void)testFlexibleSyncInitialSubscription {
+    RLMUser *user = [self logInUserForCredentials:[self basicCredentialsWithName:NSStringFromSelector(_cmd)
+                                                                        register:YES
+                                                                             app:self.flexibleSyncApp]
+                                              app:self.flexibleSyncApp];
+    RLMRealmConfiguration *config = [user flexibleSyncConfigurationWithInitialSubscriptions:^(RLMSyncSubscriptionSet *subscriptions) {
+        [subscriptions addSubscriptionWithClassName:Person.className
+                                   subscriptionName:@"person_age"
+                                              where:@"TRUEPREDICATE"];
+    } rerunOnOpen:false];
+    config.objectClasses = @[Person.self];
+    RLMRealm *realm = [RLMRealm realmWithConfiguration:config error:nil];
+    XCTAssertEqual(realm.subscriptions.count, 1UL);
+}
+
+- (void)testFlexibleSyncInitialSubscriptionAwait {
+    bool didPopulate = [self populateData:^(RLMRealm *realm) {
+        [self createPeople:realm partition:_cmd];
+    }];
+    if (!didPopulate) {
+        return;
```

Comment:
This seems like it should be a failure.

Reply:
@tgoyne added this to every flexible sync in case the test fails to setup it will not hang the CI. If you take a look at all the other test this is done on every test